### PR TITLE
JFG-1103: refactor JTestDefinition.withLoadBalancer()

### PR DESCRIPTION
**Pull request by anton-antonenko**

### CODE REVIEW

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java [hunk 1]
```diff
@@ -63,7 +66,7 @@ public JLoadScenario exampleJaggerLoadScenario() {
                 .withComment(testDefinitionComment)
                 .withInvoker(ExampleCustomHttpInvokerProvider.nonVerbose())
                 .withQueryProvider(new ExampleQueriesProvider())
-                .withLoadBalancer(new RoundRobinLoadBalancer())
+                .withLoadBalancerProvider(LoadBalancerProvider.ofRandomized(ROUND_ROBIN))
```
Comment (dlatnikov) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: propose to make
- of(DefaultLoadBalancer  defaultLoadBalancer, long seed)
- of(DefaultLoadBalancer  defaultLoadBalancer)
- of(QueryPoolLoadBalancer customLoadBalancer)

Comment (dlatnikov) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: propose rename LoadBalancerProvider to DefaultLoadBalancerProvider

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/auxiliary/Seed.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.griddynamics.jagger.user.test.configurations.auxiliary;
+
+/**
+ * Represents a Seed for Randomized Load Balancer (look at {@link LoadBalancerProvider#ofRandomized(LoadBalancerProvider.DefaultLoadBalancer, Seed)}).
+ */
+public final class Seed {
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/auxiliary/Seed.java]: do we really need it

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java (hunk 1 above)
Comment (anton-antonenko) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: It also provides custom load balancers

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/auxiliary/Seed.java (hunk 2 above)
Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/auxiliary/Seed.java]: No, but we have such classes everywhere in test builders api

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java (hunk 1 above)
Comment (anton-antonenko) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: How will you distinguish randomized and regular load balancer? 
IMO, methods with different names make it more clear for user.

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java [hunk 3]
```diff
@@ -63,7 +66,7 @@ public JLoadScenario exampleJaggerLoadScenario() {
                 .withComment(testDefinitionComment)
                 .withInvoker(ExampleCustomHttpInvokerProvider.nonVerbose())
                 .withQueryProvider(new ExampleQueriesProvider())
-                .withLoadBalancer(new RoundRobinLoadBalancer())
+                .withLoadBalancer(JLoadBalancer.builder(ROUND_ROBIN).build())
```
Comment (dlatnikov) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: please, add withRandomSeed to this example

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/loadbalancer/JLoadBalancer.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+         * Creates {@link Builder} of load balancer
+         *
+         * @param loadBalancer load balancer to use (look at {@link DefaultLoadBalancer})
+         */
+        public Builder(DefaultLoadBalancer loadBalancer) {
+            this.loadBalancer = loadBalancer;
+        }
+
+        /**
+         * Calling this setter will produce Randomized load balancer (look at {@link RandomLoadBalancer})
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/loadbalancer/JLoadBalancer.java]: Optional:

**Pull request merged**